cpp/src/arrow/compute/api_scalar.cc

```diff
@@ -858,6 +858,12 @@ Result<Datum> MapLookup(const Datum& arg, MapLookupOptions options, ExecContext*
   return CallFunction("map_lookup", {arg}, &options, ctx);
 }
 
+// ----------------------------------------------------------------------
+// Hash functions
+Result<Datum> FastHash64(const Datum& input_array, ExecContext* ctx) {
+  return CallFunction("fast_hash_64", {input_array}, ctx);
```

Comment:
The prefix "fast" doesn't add any useful information here, and you'll be embarassed if you want to propose another faster hash function. Let's just call it "hash_64"?